### DISCUSSION TITLE
When you talk about mega-editor utility, what do you mean?

### DISCUSSION THREAD

**Discussion by salazarbarrera** [AprilSylph/XKit-Rewritten#1057] [Help Desk]
Hi! I saw the changelog for the Beta 0.21.6, and saw the summary "Mega-Editor Utility: Pass errors correctly".

I saw the pull request, read #971, #969 and  #1046, and I don't really understand what that change does in concrete.

**Comment by nightpool** [comment]
This was a behind-the-scenes change that doesn't immediately change anything for users, but it lays the groundwork for a better user experience going forward by being able to understand and handle different edge case error conditions and communicate that information more effectively to the user. The first part of that change (hooking up the logic internally to be able to notice errors) was included in Beta 0.21.6, but the second part of that change hasn't been completed yet.

**Reply by marcustyphoon** [reply]
Yep! Before #971, scripts could not perform error-handling logic on mega-editor errors because they would not be informed that anything went wrong. With that PR merged, they _can_ do this, but they currently don't.
